transportable-udfs-compile-utils/src/main/java/com/linkedin/transport/compile/UDFProperties.java

```diff
@@ -22,18 +23,27 @@
   private Multimap<String, String> _udfs;
 
   public UDFProperties() {
-    _udfs = HashMultimap.create();
+    _udfs = LinkedHashMultimap.create();
   }
 
-  void addUDF(String topLevelStdUDFClassName, String udfClassName) {
+  public void addUDF(String topLevelStdUDFClassName, String udfClassName) {
     _udfs.put(topLevelStdUDFClassName, udfClassName);
   }
 
+  public Multimap<String, String> getUdfs() {
+    return _udfs;
+  }
+
   public void toJson(Writer writer) {
     Gson gson = new GsonBuilder().setPrettyPrinting().create();
     gson.toJson(UDFPropertiesFile.fromUDFProperties(this), writer);
   }
 
+  public static UDFProperties fromJson(Reader reader) {
+    Gson gson = new GsonBuilder().create();
+    return gson.fromJson(reader, UDFPropertiesFile.class).toUDFProperties();
```

Comment:
Can we get rid of UDFPropertiesFile? I think it is very simple to get the Json map and convert it to multimap. Creating two classes just for that (UDF and UDFPropertiesFile) sounds like an overkill.